lib/util/trip/ride_card.dart

```diff
@@ -92,7 +92,42 @@ class _RideCardState extends TripCardState<Ride, RideCard> {
 
   @override
   Widget buildTopRight() {
-    return Text(key: const Key('price'), ' ${_ride.price}€');
+    if (!_fullyLoaded) {
+      return const Center(
+        child: SizedBox(),
+      );
+    } else {
+      switch (_ride.status) {
+        case RideStatus.preview:
+          return Text(key: const Key('price'), ' ${_ride.price}€');
+        case RideStatus.pending:
+          return Icon(
+            Icons.access_time_outlined,
+            color: pickStatusColor(),
+            key: const Key('pending'),
+          );
+        case RideStatus.approved:
+          return Icon(
+            Icons.done_all,
+            color: pickStatusColor(),
+            key: const Key('approved'),
+          );
+        case RideStatus.rejected:
+          return Icon(
+            Icons.close,
```

Comment:
Icons.block makes more sense here, and use something else for cancellation?

Reply:
I think its easiest for all cancelled/rejected rides to have the same icon